### PR TITLE
Fix async navigation in a couple places

### PR DESCRIPTION
**Pull request by pt2121**
re: #332

Fix async navigation in a couple places:
* Analysis
* Liked Songs
* Recently Played

### CODE REVIEW

**Comment by ericonr** [comment]
Just tested it out on my system! Worked great, and I think the screen looked pretty good before loading as well.

Awesome fix :)

My internet connection is currently a bit spotty, so I will get to stress test it quite a bit :p

File: src/network.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ pub enum IoEvent {
   SetTracksToTable(Vec<FullTrack>),
   GetMadeForYouPlaylistTracks(String, u32),
   GetPlaylistTracks(String, u32),
-  GetCurrentSavedTracks(Option<u32>, bool),
```
Comment (Rigellute) [inline src/network.rs]: Good work removing `bool` param, I should not have added it in the first place 😉

**Comment by Rigellute** [comment]
Will merge once conflicts are resolved. Are you able to rebase?

**Comment by pt2121** [comment]
Rebased. Thanks!

**Comment by Rigellute** [comment]
Thanks again 👍 

@all-contributors please add @pt2121 for code

**Pull request merged**